### PR TITLE
Fix precision recognition

### PR DESCRIPTION
**Pull request by vvkh**
In Postgres, if type is specified as `NUMERIC` (no scale/precision parameters), the effective scale is 16383 ([docs](https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-TABLE)).
In BigQuery, the effective scale in this case is 9 ([community forum](https://www.googlecloudcommunity.com/gc/Databases/What-s-default-Scale-and-Precision-of-Numeric-type-in-BigQuery/m-p/612292#M1281)).

### CODE REVIEW

File: data_diff/databases/bigquery.py [hunk 1]
```diff
@@ -76,6 +76,11 @@ class Dialect(BaseDialect):
     }
     TYPE_ARRAY_RE = re.compile(r"ARRAY<(.+)>")
     TYPE_STRUCT_RE = re.compile(r"STRUCT<(.+)>")
+    # [BIG]NUMERIC, [BIG]NUMERIC(precision, scale), [BIG]NUMERIC(precision)
+    TYPE_NUMERIC_RE = re.compile(r"^((BIG)?NUMERIC)(?:\((\d+)(?:, (\d+))?\))?$")
```
Comment (dlawin) [inline data_diff/databases/bigquery.py:80]: [BIG]NUMERIC() not possible?

**Review by dlawin — APPROVED** [review]

File: data_diff/databases/bigquery.py (hunk 1 above)
Comment (vvkh) [reply data_diff/databases/bigquery.py:80]: Yep, it's invalid in BQ.
image

Comment (nolar) [inline data_diff/databases/bigquery.py:80]: Is it guaranteed to have strictly 1 space after the comma? Never 0?

Comment (vvkh) [reply data_diff/databases/bigquery.py:80]: Yes, the space is always there even if it user didn't add it explicitly.

**Pull request merged**